app/javascript/redux/greetSlice/greetSlice.js

```diff
@@ -0,0 +1,54 @@
+import axios from "axios";
+
+const { createAsyncThunk, createSlice } = require("@reduxjs/toolkit");
+
+export const fetchGreeting = createAsyncThunk(
+  'greet/fetchGreetings',
+  async () => {
+    try {
+      let response = await axios('http://127.0.0.1:5000/api/greetings')
+      const data = response.data
+      return data.greeting
+    }
```

Comment:
The Front-end app is running fine and routing works fine as expected, but:
- [ ] The URL used here, when fetching greetings, is not returning greetings as expected. Please look at the errors I got on the console when I inspected the code. 👍🏽
- [ ] The server that is expected to return greetings from the API is not working as it should. Kindly fix that so random greetings are returned and are displayed on the UI as expected 👍🏽

<img width="912" alt="Screenshot 2024-02-12 at 5 17 27 PM" src="https://github.com/hamidazim321/hello-rails-react/assets/50941074/1aab443f-766c-4642-abe5-fe998e939ee7">

Reply:
hello @AmaduKamara , the error you are getting is probably because you opened the browser on `http://localhost:5000/greetings` while the server is running on `http://127.0.0.1:5000/greetings`. This is a security feature that prevents requests to your server from scripts running on pages with a different origin (domain, protocol, or port). Please try opening the browser on `http://127.0.0.1:5000/greetings` URL and it will work fine. I get the same error when I open it on the wrong URL `http://localhost:5000/greetings`.

Reply:
@AmaduKamara here's the difference when opening the project on those URLs:

`http://localhost:5000/greetings`:
![p](https://github.com/hamidazim321/hello-rails-react/assets/129533107/2103d2cc-68b7-449e-9fc7-25039edb0c7c)

`http://127.0.0.1:5000/greetings`:
![s](https://github.com/hamidazim321/hello-rails-react/assets/129533107/34a71e81-610a-4616-8af2-f5d9671f2e16)

Only getting errors on `http://localhost:5000/greetings` because the domain doesn't match the domain on which the server is running

Reply:
I have tried both URLs and I get the same error.